does not compile on android

compilation of master fails on android:

<pre>
/home/jenkins/workspace/Android-ARM/tools/depends/target/binary-addons/arm-linux-androideabi-android-21/build/pvr.mediaportal.tvserver/src/lib/live555/groupsock/GroupsockHelper.cpp:503:21: error: request for member 's_addr' in 'imr.ip_mreq_source::imr_multiaddr', which is of non-class type '__be32 {aka unsigned int}'
   imr.imr_multiaddr.s_addr = groupAddress;
                     ^
[ 64%] Building C object CMakeFiles/pvr.mediaportal.tvserver.dir/src/lib/live555/groupsock/inet.c.o
/home/jenkins/workspace/Android-ARM/tools/depends/target/binary-addons/arm-linux-androideabi-android-21/build/pvr.mediaportal.tvserver/src/lib/live555/groupsock/GroupsockHelper.cpp:504:22: error: request for member 's_addr' in 'imr.ip_mreq_source::imr_sourceaddr', which is of non-class type '__be32 {aka unsigned int}'
   imr.imr_sourceaddr.s_addr = sourceFilterAddr;
                      ^
/home/jenkins/workspace/Android-ARM/tools/depends/target/binary-addons/arm-linux-androideabi-android-21/build/pvr.mediaportal.tvserver/src/lib/live555/groupsock/GroupsockHelper.cpp:505:21: error: request for member 's_addr' in 'imr.ip_mreq_source::imr_interface', which is of non-class type '__be32 {aka unsigned int}'
   imr.imr_interface.s_addr = ReceivingInterfaceAddr;
                     ^
/home/jenkins/workspace/Android-ARM/tools/depends/target/binary-addons/arm-linux-androideabi-android-21/build/pvr.mediaportal.tvserver/src/lib/live555/groupsock/GroupsockHelper.cpp: In function 'Boolean socketLeaveGroupSSM(UsageEnvironment&, int, netAddressBits, netAddressBits)':
/home/jenkins/workspace/Android-ARM/tools/depends/target/binary-addons/arm-linux-androideabi-android-21/build/pvr.mediaportal.tvserver/src/lib/live555/groupsock/GroupsockHelper.cpp:521:21: error: request for member 's_addr' in 'imr.ip_mreq_source::imr_multiaddr', which is of non-class type '__be32 {aka unsigned int}'
   imr.imr_multiaddr.s_addr = groupAddress;
                     ^
/home/jenkins/workspace/Android-ARM/tools/depends/target/binary-addons/arm-linux-androideabi-android-21/build/pvr.mediaportal.tvserver/src/lib/live555/groupsock/GroupsockHelper.cpp:522:22: error: request for member 's_addr' in 'imr.ip_mreq_source::imr_sourceaddr', which is of non-class type '__be32 {aka unsigned int}'
   imr.imr_sourceaddr.s_addr = sourceFilterAddr;
                      ^
/home/jenkins/workspace/Android-ARM/tools/depends/target/binary-addons/arm-linux-androideabi-android-21/build/pvr.mediaportal.tvserver/src/lib/live555/groupsock/GroupsockHelper.cpp:523:21: error: request for member 's_addr' in 'imr.ip_mreq_source::imr_interface', which is of non-class type '__be32 {aka unsigned int}'
   imr.imr_interface.s_addr = ReceivingInterfaceAddr;
</pre>

--- a/pvr.mediaportal.tvserver/addon.xml.in
+++ b/pvr.mediaportal.tvserver/addon.xml.in
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <addon
   id="pvr.mediaportal.tvserver"
-  version="2.4.6"
+  version="2.4.7"
   name="MediaPortal PVR Client"
   provider-name="Marcel Groothuis">
   <requires>

--- a/pvr.mediaportal.tvserver/changelog.txt
+++ b/pvr.mediaportal.tvserver/changelog.txt
@@ -1,3 +1,6 @@
+v2.4.7
+- Fixed: Live555: Android compile fix
+
 v2.4.6
 - Fixed: various Coverity reported issues for Live555
 

--- a/src/lib/live555/groupsock/GroupsockHelper.cpp
+++ b/src/lib/live555/groupsock/GroupsockHelper.cpp
@@ -500,9 +500,15 @@ Boolean socketJoinGroupSSM(UsageEnvironment& env, int socket,
   if (!IsMulticastAddress(groupAddress)) return True; // ignore this case
 
   struct ip_mreq_source imr;
-  imr.imr_multiaddr.s_addr = groupAddress;
-  imr.imr_sourceaddr.s_addr = sourceFilterAddr;
-  imr.imr_interface.s_addr = ReceivingInterfaceAddr;
+#ifdef __ANDROID__
+    imr.imr_multiaddr = groupAddress;
+    imr.imr_sourceaddr = sourceFilterAddr;
+    imr.imr_interface = ReceivingInterfaceAddr;
+#else
+    imr.imr_multiaddr.s_addr = groupAddress;
+    imr.imr_sourceaddr.s_addr = sourceFilterAddr;
+    imr.imr_interface.s_addr = ReceivingInterfaceAddr;
+#endif
   if (setsockopt(socket, IPPROTO_IP, IP_ADD_SOURCE_MEMBERSHIP,
 		 (const char*)&imr, sizeof (struct ip_mreq_source)) < 0) {
     socketErr(env, "setsockopt(IP_ADD_SOURCE_MEMBERSHIP) error: ");
@@ -518,9 +524,15 @@ Boolean socketLeaveGroupSSM(UsageEnvironment& /*env*/, int socket,
   if (!IsMulticastAddress(groupAddress)) return True; // ignore this case
 
   struct ip_mreq_source imr;
-  imr.imr_multiaddr.s_addr = groupAddress;
-  imr.imr_sourceaddr.s_addr = sourceFilterAddr;
-  imr.imr_interface.s_addr = ReceivingInterfaceAddr;
+#ifdef __ANDROID__
+    imr.imr_multiaddr = groupAddress;
+    imr.imr_sourceaddr = sourceFilterAddr;
+    imr.imr_interface = ReceivingInterfaceAddr;
+#else
+    imr.imr_multiaddr.s_addr = groupAddress;
+    imr.imr_sourceaddr.s_addr = sourceFilterAddr;
+    imr.imr_interface.s_addr = ReceivingInterfaceAddr;
+#endif
   if (setsockopt(socket, IPPROTO_IP, IP_DROP_SOURCE_MEMBERSHIP,
 		 (const char*)&imr, sizeof (struct ip_mreq_source)) < 0) {
     return False;

